### PR TITLE
Add rootReset boolean to distinguish root ops

### PR DESCRIPTION
**Pull request by alshakero**

### CODE REVIEW

File: src/json-patch-duplex.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ module jsonpatch {
     removed?: any,
     test?: boolean,
     newDocument: T;
+    rootReset: Boolean;
```
Comment (felixfbecker) [inline src/json-patch-duplex.ts]: scalar types are lowercased (like the returns of `typeof`)

**Comment by felixfbecker** [comment]
Isn't this equivalent to `result.newDocument === oldDocument`?

**Comment by alshakero** [comment]
@felixfbecker not if you replace root.

**Comment by felixfbecker** [comment]
Sorry, meant `result.newDocument !== oldDocument`. That should be `true` if the root was replaced.

**Comment by alshakero** [comment]
@felixfbecker Oops, that didn't cross my mind.

Hmmm. Even though I see the use case for this boolean, I guess adding an extra `if` in Palindrom is better. Since this lib is all about speed.

Closing.

**Pull request closed without merging**